.github/workflows/build-artifact.yaml

```diff
@@ -2,13 +2,19 @@ name: "Build tt-metal artifacts"
 
 on:
   workflow_call:
+    inputs:
+      arch:
+        required: false
+        type: string
+        default: '["grayskull", "wormhole_b0"]'
```

Comment:
Is it possible to instead make `build-artifact` take in only one arch as a string, and then change any callers to specify a matrix for `build-artifacts`?

So caller workflow would be in charge of the matrix-ing, not `build-artifact`. You just pass `${{ matrix.arch }}`` or whatever into `arch: ` for `build-artifacts`.

Not sure if that's possible with using reusable workflows...

Reply:
Not quite sure the exact change you're imaginging so will answer piece by piece and hope that answers it all.
with currently changes, it is possible to specify only 1 arch to build with a build-artifacts.

for workflows which will call the build-artifact workflow, i believe the strategy.matrix.arch setup is only on a per job basis. I don't think you can set a strategy that encompasses multiple jobs, a strategy must be manually set for each job.

If we changed build-artifact into a GH action I think we can then programmatically feed in "arch" since we can then write it as a single job. but then we'd be running this on either a full BM or VM process.